packages/dashboard/src/components/iot-dashboard/iot-dashboard.tsx

```diff
@@ -0,0 +1,272 @@
+  }
+
+  return { x: totalOffsetX, y: totalOffsetY };
+};
+
+@Component({
+  tag: 'iot-dashboard',
+  styleUrl: 'iot-dashboard.css',
+  shadow: false,
+})
+export class IotDashboard {
+  /** The configurations which determines which widgets render where with what settings. */
+  @Prop() dashboardConfiguration: DashboardConfiguration;
+
+  /**
+   * Callback that is fired every time the dashboard configuration has been altered.
+   *
+   * When a widget is moved, resized, deleted, appended, or altered, then this method is called
+   */
+  @Prop() onDashboardConfigurationChange: (config: DashboardConfiguration) => void;
+
+  /**
+   * Whether the dashboard grid will stretch to fit.
+   *
+   * If stretch to fit is false, the dashboard grid will be the width in pixels.
+   * If not enough room is present, it will utilize scrollbars to allow access to the entire grid.
+   *
+   * If stretch to fit is true, the entire grid will scale proportionally to scale to the available space for the grid.
+   */
+  @Prop() stretchToFit: Boolean = DEFAULT_STRETCH_TO_FIT;
+
+  /** Width of the dashboard, in pixels */
+  @Prop() width: number;
+
+  /** Width and height of the cell, in pixels */
+  @Prop() cellSize: number = DEFAULT_CELL_SIZE;
+
+  /** List of ID's of the currently selected widgets. */
+  @State() selectedWidgetIds: string[] = [];
+
+  /** The dashboard configurations current state. This is what the dashboard reflects as truth. */
+  @State() currDashboardConfiguration: DashboardConfiguration;
+
+  /** Selection gesture */
+  @State() start: Position | undefined;
+  @State() end: Position | undefined;
+  @State() finishedSelecting: boolean = false;
+  @State() previousPosition: Position | undefined;
+
+  /** Resize gesture */
+  @State() resizingInfo: ResizingInfo | undefined = undefined;
+
+  @State() currWidth: number;
+  @Element() el!: HTMLElement;
+
+  private resizer: ResizeObserver;
+
+  componentWillLoad() {
+    this.currDashboardConfiguration = this.dashboardConfiguration;
+
+    /**
+     * Creates a listener for elements dimensions changing.
+     * This allows us to dynamically set the widget dimensions.
+     */
+    this.resizer = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        /** Update Size */
+        const { width } = entry.contentRect;
+        this.currWidth = width;
+      });
+    });
+
+    this.currWidth = this.width;
+  }
+
+  componentDidLoad() {
+    this.resizer.observe(this.el.firstElementChild as Element);
+  }
+
+  @Watch('dashboardConfiguration')
+  watchDashboardConfiguration(newDashboardConfiguration: DashboardConfiguration) {
+    this.currDashboardConfiguration = newDashboardConfiguration;
+  }
+
+  resize = (position: Position, side: WidgetSide, widgetId: string) => {
+    this.resizingInfo = { side, widgetId };
+    this.previousPosition = position;
+  };
+
+  onStart({ x, y }: Position) {
+    console.log('on start');
```

Comment:
😎

Reply:
hm i think there's something astray with our eslint enforcement, should have caught this. will look into that